sklearn/model_selection/tests/test_split.py

```diff
@@ -430,19 +430,30 @@ def test_shuffle_kfold_stratifiedkfold_reproducibility():
         np.testing.assert_equal(list(cv.split(X, y)), list(cv.split(X, y)))
         np.testing.assert_equal(list(cv.split(X2, y2)), list(cv.split(X2, y2)))
 
+    # Tests to ensure consistent behaviour even when random_state is not set.
     kf = KFold(3, shuffle=True)
     skf = StratifiedKFold(3, shuffle=True)
-
-    for cv in (kf, skf):
+    kf1 = KFold(3, shuffle=True)
+    kf2 = KFold(3, shuffle=True)
+    skf1 = StratifiedKFold(3, shuffle=True)
+    skf2 = StratifiedKFold(3, shuffle=True)
+    for cvs in ((kf, kf1, kf2), (skf, skf1, skf2)):
         for data in zip((X, X2), (y, y2)):
+            # For the same initialization, splits should be same across
+            # multiple split calls, even when random_state is not set.
+            np.testing.assert_equal(list(cvs[0].split(*data)),
+                                    list(cvs[0].split(*data)))
+
+            # For different initialisations, splits should not be same when
+            # random_state is not set.
             try:
-                np.testing.assert_equal(list(cv.split(*data)),
-                                        list(cv.split(*data)))
+                np.testing.assert_equal(list(cvs[1].split(*data)),
```

Comment:
I was going by the current numpy implementation, which I guess you found a bug in?